### PR TITLE
Better support for `cls` inheritance

### PR DESCRIPTION
**Pull request by aksh-at**
This now works:

```python
class Base:
    def __enter__(self):
        self.x = 2

    @modal.method()
    def run(self, y):
        return self.x * y

@stub.cls()
class Foo(Base):
    pass

@stub.local_entrypoint()
def main():
    print(Foo().run.remote(3)) # == 6
```

### CODE REVIEW

File: modal/_function_utils.py [hunk 1]
```diff
@@ -86,8 +86,17 @@ class FunctionInfo:
...
         self.raw_f = f
         self.cls = cls
-        # TODO(erikbern): if f.__qualname__ != f.__name__,  we should infer the class name instead
-        self.function_name = name_override if name_override is not None else f.__qualname__
+
+        if name_override is not None:
+            self.function_name = name_override
+        elif f.__qualname__ != f.__name__:
+            # Class function.
+            if len(f.__qualname__.split(".")) > 2:
```
Comment (aksh-at) [inline modal/_function_utils.py:94]: maybe we should fallback to `__qualname__` here if len > 2? I don't think anyone was legitimately using that though

File: modal/stub.py [hunk 2]
```diff
@@ -609,11 +609,14 @@ def wrapper(user_cls: CLS_T) -> _Cls:
             partial_functions: Dict[str, PartialFunction] = {}
             functions: Dict[str, _Function] = {}
 
-            for k, v in user_cls.__dict__.items():
-                if isinstance(v, PartialFunction):
-                    partial_functions[k] = v
-                    partial_function = synchronizer._translate_in(v)  # TODO: remove need for?
-                    functions[k] = decorator(partial_function, user_cls)
+            for parent_cls in user_cls.mro():
```
Comment (erikbern) [inline modal/stub.py:612]: neat, didn't know you can do this

**Comment by erikbern** [comment]
we should add a test to `container_test.py` too – i'm happy to do that if it's not straightforward. that makes sure it works on the container side (importing functions and executing them)

**Review by erikbern — APPROVED** [review]
great with container tests too!

**Pull request merged**